test/assets/unrunnableTestCases.csv

```diff
@@ -3116,145 +3093,47 @@ orderBy63,Error: Not Implemented: default namespace element.
-K2-OrderbyExprWithout-35,AssertionError: Expected executing the XPath "let $i := (xs:hexBinary("FF"), xs:time("03:03:03Z"), xs:hexBinary("FF")) stable order by $i[1] return $i" to resolve to one of the expected results, but got Error: Not implemented: orderByClause is not implemented yet., AssertionError: expected [Function] to throw error matching /XPTY0004/ but got 'Not implemented: orderByClause is not implemented yet.'.
-K2-OrderbyExprWithout-36,AssertionError: Expected executing the XPath "let $i := (xs:hexBinary("FF"), xs:time("03:03:03Z"), xs:hexBinary("FF")) order by $i[1] return $i" to resolve to one of the expected results, but got Error: Not implemented: orderByClause is not implemented yet., AssertionError: expected [Function] to throw error matching /XPTY0004/ but got 'Not implemented: orderByClause is not implemented yet.'.
-K2-OrderbyExprWithout-37,AssertionError: Expected executing the XPath "for $i in (1, 3, 2) stable order by () return $i" to resolve to one of the expected results, but got Error: Not implemented: orderByClause is not implemented yet., Error: Not implemented: orderByClause is not implemented yet., AssertionError: expected [Function] to throw error matching /XPST0005/ but got 'Not implemented: orderByClause is not implemented yet.'.
-K2-OrderbyExprWithout-38,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-39,AssertionError: Expected executing the XPath "for $a in (1, 4, 2) let $i := (1, 3, 2) order by $i return 1" to resolve to one of the expected results, but got Error: Not implemented: orderByClause is not implemented yet., AssertionError: expected [Function] to throw error matching /XPTY0004/ but got 'Not implemented: orderByClause is not implemented yet.'.
-K2-OrderbyExprWithout-40,AssertionError: expected [Function] to throw error matching /XPTY0004/ but got 'Not implemented: orderByClause is not implemented yet.'
-K2-OrderbyExprWithout-41,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-42,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-43,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-44,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-45,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-46,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-47,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-48,Error: Not implemented: orderByClause is not implemented yet.
-K2-OrderbyExprWithout-49,Error: Not implemented: orderByClause is not implemented yet.
-Parenexpr-11,Error: Not implemented: orderByClause is not implemented yet.
+K2-OrderbyExprWithout-46,Error: XPTY0004: Could not cast values
```

Comment:
@ThomasGravekamp could you take a look at this test? I do not really expect it to still fail

Reply:
It is this one: ```   <test-case name="K2-OrderbyExprWithout-46">
      <description> Sort special floating point values. </description>
      <created by="Frans Englich" on="2007-11-26"/>
      <test>let $numbers := (1, 2, 1.3, 3e3, xs:double("NaN"), xs:double("-INF"), xs:double("INF")) return (for $i in $numbers order by $i empty least return $i, "SEP", for $i in $numbers order by $i empty greatest return $i)</test>
      <result>
         <assert-string-value>NaN -INF 1 1.3 2 3000 INF SEP -INF 1 1.3 2 3000 INF NaN</assert-string-value>
      </result>
   </test-case>
```

Reply:
Hm, never mind, it must be one of the NaN / INF versions, which is unrelated to this PR.

Reply:
Although it could be this one: https://github.com/FontoXML/fontoxpath/pull/276/files/34cf1dab2ddc04f34c685e4bafe057bf621a6b83#diff-6efeeca118484e692dd293a509562ec66d3146494439df5d2b9aba5cf495f9f4, which should not be hit in this case